src/core/control/xojfile/LoadHandler.cpp

```diff
@@ -351,10 +351,10 @@ void LoadHandler::addStroke(StrokeTool tool, const Color& color, double width, i
         this->stroke->setLineStyle(*lineStyle);
     }
 
-    setAudioAttributes(*this->stroke, filename, timestamp);
+    setAudioAttributes(*this->stroke, std::move(filename), timestamp);
 }
 
-void LoadHandler::setStrokePoints(std::vector<Point>&& pointVector, std::vector<double> pressures) {
+void LoadHandler::setStrokePoints(std::vector<Point> pointVector, std::vector<double> pressures) {
```

Comment:
C++ question: having `std::vector<Point>&& pointVector` as argument, does not tell the caller the intention that the vector will be moved? This also avoids the potential pitfall at call site to make a copy.

Reply:
I was confused as well, but there is a good explanation from @Febbe for this in the 15th comment of his review

Reply:
Are you referencing to [this comment](https://github.com/xournalpp/xournalpp/pull/5554#discussion_r1678514222)? I agree with @Febbe  but I would like to rephrase what I understood :

 - small non-owning-any-resource objects can be safely copied instead of being moved or referenced, so `const Color& color` could be replaced by `const Color color` (I prefer to keep the `const` just to signal the compiler that if I try to modify the argument it's due a mistake). The reason is performance, copying small stuff is cheap.
 - large objects or objects that owns-a-resource (like `unique_ptr`) needs to be either moved or referenced (depending if the callee wants to own or not the object). For the former case, `std::vector<Point> pointVector` should become `std::vector<Point>&& pointVector`, for the latter, `std::vector<Point> pointVector` should become `std::vector<Point>& pointVector`. 
 
It is true that a parameter declared as `std::vector<Point> pointVector` not necessary triggers a copy: the caller can still invoke the function as `fun(std::move(vector))` and the move constructor should kick in. But if the caller unintentionally calls `fun(vector)`, you will get a unnecessary copy.
My preference is to prevent this declaring the parameter as `std::vector<Point>&& pointVector`, not because you can avoid a call to the move constructor but because you will catch unintentional copies.

One fair point that @Febbe  makes is that calling `std::move` or receiving `foo&& arg` does not trigger a real move and the callee may change the original object without moving it.
Sad fact.
In theory, `clang-tidy` should [catch this for us](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/rvalue-reference-param-not-moved.html).

Reply:
I see your point, but I think he still generally advised against it. Moving an object _may_ offer better performance than passing a reference, and it is safer in async scenarios (which is a possibility for the future of this code). But I'm not an expert on this either... Maybe we can have a second clarification from @Febbe?

Reply:
Sorry for the confusion. 

Both variants have their problems and benefits.
Its always a tradeoff and whether to use "pass by rvalue reference" or "pass by value" really depends on the situation. Unfortunately we don't have any explicit pass by value in c++.

Nevertheless, as I wrote before, using rvalue references is fine or even advised, when you want to force the usage to be explicit. This is the case, when you really know, that the caller does not need this value, or if you know, that implicit ops might cause severe problems. This is the case for large point vectors. But it is not for strings(which are often used at several places).

The reason against this is, that pass by value is more clear, simple and readable. It also pulls the place, where a copy or move happens to the caller side. On top, constructors usually also take parameters, which are moved by value. That's why I prefer pass by value over pass by rvalue reference. 

Also, when using coroutines, references must not be used, but that's not the case here.

@tmoerchel you can leave this as is for now. We can still decide, if we want to refactor such cases later in the whole software.

Regarding clang-tidy, it can also catch movable copies and advise to be explicit, when you really want to do a copy.